pmd-java/src/main/java/net/sourceforge/pmd/lang/java/metrics/impl/visitors/DefaultNpathVisitor.java

```diff
@@ -0,0 +1,194 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.java.metrics.impl.visitors;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import net.sourceforge.pmd.lang.java.ast.ASTConditionalExpression;
+import net.sourceforge.pmd.lang.java.ast.ASTDoStatement;
+import net.sourceforge.pmd.lang.java.ast.ASTExpression;
+import net.sourceforge.pmd.lang.java.ast.ASTForStatement;
+import net.sourceforge.pmd.lang.java.ast.ASTIfStatement;
+import net.sourceforge.pmd.lang.java.ast.ASTMethodOrConstructorDeclaration;
+import net.sourceforge.pmd.lang.java.ast.ASTReturnStatement;
+import net.sourceforge.pmd.lang.java.ast.ASTStatement;
+import net.sourceforge.pmd.lang.java.ast.ASTSwitchLabel;
+import net.sourceforge.pmd.lang.java.ast.ASTSwitchStatement;
+import net.sourceforge.pmd.lang.java.ast.ASTTryStatement;
+import net.sourceforge.pmd.lang.java.ast.ASTWhileStatement;
+import net.sourceforge.pmd.lang.java.ast.JavaNode;
+import net.sourceforge.pmd.lang.java.ast.JavaParserVisitorReducedAdapter;
+import net.sourceforge.pmd.lang.java.metrics.impl.CycloMetric;
+
+/**
+ * Visitor for the default n-path complexity version.
+ *
+ * @author Clément Fournier
+ * @author Jason Bennett
+ */
+public class DefaultNpathVisitor extends JavaParserVisitorReducedAdapter {
+
+    /* Multiplies the complexity of the children of this node. */
+    private int multiplyChildrenComplexities(JavaNode node, Object data) {
+        int product = 1;
+
+        for (int i = 0; i < node.jjtGetNumChildren(); i++) {
+            JavaNode n = (JavaNode) node.jjtGetChild(i);
+            product *= (Integer) n.jjtAccept(this, data);
+        }
+
+        return product;
+    }
+
+
+    /* Sums the complexity of the children of the node. */
+    private int sumChildrenComplexities(JavaNode node, Object data) {
+        int sum = 0;
+
+        for (int i = 0; i < node.jjtGetNumChildren(); i++) {
+            JavaNode n = (JavaNode) node.jjtGetChild(i);
+            sum += (Integer) n.jjtAccept(this, data);
+        }
+
+        return sum;
+    }
+
+
+    @Override
+    public Object visit(ASTMethodOrConstructorDeclaration node, Object data) {
+        return multiplyChildrenComplexities(node, data);
+    }
+
+
+    @Override
+    public Object visit(JavaNode node, Object data) {
+        return multiplyChildrenComplexities(node, data);
+    }
+
+
+    @Override
+    public Object visit(ASTIfStatement node, Object data) {
+        // (npath of if + npath of else (or 1) + bool_comp of if) * npath of next
+
+        List<JavaNode> statementChildren = new ArrayList<>();
+        for (int i = 0; i < node.jjtGetNumChildren(); i++) {
+            if (node.jjtGetChild(i) instanceof ASTStatement) {
```

Comment:
Would it work here to use `findChildrenOfType(ASTStatement.class)` instead of checking the type directly?